policy-controller/src/main.rs

```diff
@@ -110,22 +106,41 @@ async fn main() -> Result<()> {
         let (handle, index) =
             linkerd_policy_controller::k8s::Index::new(cluster, default_policy, DETECT_TIMEOUT);
 
-        tokio::spawn(index.run(client.clone(), ready_tx));
+        tokio::spawn(index.run(client.clone(), readiness.clone()));
         handle
     };
 
     // Run the gRPC server, serving results by looking up against the index handle.
     tokio::spawn(grpc(grpc_addr, cluster_networks, handle, drain_rx.clone()));
 
-    if let Some(bind_addr) = admission_addr {
-        let (listen_addr, serve) = bind_admission_controller(bind_addr, client).await?;
-        tokio::spawn(serve.instrument(info_span!("admission")));
-        info!(addr = %listen_addr, "Admission controller server listening");
+    if admission_controller_disabled {
+        tracing::info!("Admission controller disabled");
+    } else {
+        let srv = server
+            .spawn(admission::Service { client }, drain_rx.clone())
+            .await?;
+        info!(addr = %srv.local_addr(), "Admission controller server listening");
```

Comment:
what does the future returned by `kubert::server::ServerArgs::spawn` actually represent? does that mean the server is running, or the server task has _completed_? this was not super obvious from the docs --- I'm guessing it means the server is running, but this code looked a little sketchy the first time i saw it, and i wanted to make sure this wasn't accidentally awaiting the server task _ending_

Reply:
yeah the binding happens async and can fail before the server task is actually spawned, I'll document spawn.